wgpu-core/src/binding_model.rs

```diff
@@ -498,10 +541,10 @@ pub(crate) fn get_bind_group_layout<A: HalApi>(
     id: Valid<BindGroupLayoutId>,
 ) -> (Valid<BindGroupLayoutId>, &BindGroupLayout<A>) {
     let layout = &layouts[id];
-    layout
-        .compatible_layout
-        .map(|compat| (compat, &layouts[compat]))
-        .unwrap_or((id, layout))
+    if let BglOrDuplicate::Duplicate(original_id) = layout.inner {
+        return (original_id, &layouts[original_id]);
+    }
+    (id, layout)
```

Comment:
nitpick: This diverges less from previous code, and I think it's easier to follow:

```suggestion
    layout
        .as_duplicate()
        .map_or((id, layout), |deduped| (deduped, &layouts[deduped]))
```

Pushed as a `fixup!` with 03a51250.